tx/textual/valuerenderer/valuerenderer.go

```diff
@@ -1,25 +1,43 @@
 package valuerenderer
 
 import (
+	"context"
 	"fmt"
 
 	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/reflect/protoreflect"
+	"google.golang.org/protobuf/types/known/timestamppb"
 
+	bankv1beta1 "cosmossdk.io/api/cosmos/bank/v1beta1"
+	basev1beta1 "cosmossdk.io/api/cosmos/base/v1beta1"
 	cosmos_proto "github.com/cosmos/cosmos-proto"
 )
 
+// CoinMetadataQueryFn defines a function that queries state for the coin denom
+// metadata. It is meant to be passed as an argument into `NewTextual`.
+type CoinMetadataQueryFn func(ctx context.Context, denom string) (*bankv1beta1.Metadata, error)
+
 // Textual holds the configuration for dispatching
 // to specific value renderers for SIGN_MODE_TEXTUAL.
 type Textual struct {
-	scalars  map[string]ValueRenderer
+	// coinMetadataQuerier defines a function to query the coin metadata from
+	// state.
+	coinMetadataQuerier CoinMetadataQueryFn
+	// scalars defines a registry for Cosmos scalars.
+	scalars map[string]ValueRenderer
+	// messages defines a registry for custom message renderers, as defined in
+	// point #9 in the spec. Note that we also use this same registry for the
+	// following messages, as they can be thought of custom message rendering:
+	// - SDK coin and coins
+	// - Protobuf timestamp
+	// - Protobuf duration
 	messages map[protoreflect.FullName]ValueRenderer
 }
 
 // NewTextual returns a new Textual which provides
 // value renderers.
-func NewTextual() Textual {
-	t := Textual{}
+func NewTextual(q CoinMetadataQueryFn) Textual {
+	t := Textual{coinMetadataQuerier: q}
```

Comment:
You should provide a default querier within this package and add it to the `Textual` here, just like we add all the built-in renderers. Consider adding a `NewRawTextual()` with initialization methods for use by tests and custom rendering uses.

Reply:
What would the default querier be?